packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -0,0 +1,157 @@
+  await addAuthWithDefault(projRoot);
+  await amplifyPushAuth(projRoot);
+}
+
+export async function assertGen1Setup(projRoot: string) {
+  const gen1Meta = getProjectMeta(projRoot);
+  const gen1UserPoolId = Object.keys(gen1Meta.auth).map((key) => gen1Meta.auth[key])[0].output.UserPoolId;
+  const gen1Region = gen1Meta.providers.awscloudformation.Region;
+  const userPool = await getUserPool(gen1UserPoolId, gen1Region);
+  expect(userPool.UserPool).toBeDefined();
+  return { gen1UserPoolId, gen1Region };
+}
+
+export async function assertUserPoolResource(projRoot: string, gen1UserPoolId: string, gen1Region: string) {
+  let gen1Resource = await getResourceDetails('AWS::Cognito::UserPool', gen1UserPoolId, gen1Region);
+  gen1Resource = removeNestedJsonKeys(gen1Resource, ['ProviderURL', 'ProviderName', 'UserPoolId', 'Arn']);
+  // TODO: remove below line after EmailMessage, EmailSubject, SmsMessage, SmsVerificationMessage, EmailVerificationMessage, EmailVerificationSubject, AccountRecoverySetting inconsistency is fixed
+  gen1Resource = removeNestedJsonKeys(gen1Resource, [
+    'UserPoolTags',
+    'VerificationMessageTemplate.EmailMessage',
+    'VerificationMessageTemplate.EmailSubject',
+    'EmailVerificationSubject',
+    'AccountRecoverySetting',
+    'EmailVerificationMessage',
+  ]);
+  const gen2Meta = getProjectOutputs(projRoot);
+  const gen2UserPoolId = gen2Meta.auth.user_pool_id;
+  const gen2Region = gen2Meta.auth.aws_region;
+  let gen2Resource = await getResourceDetails('AWS::Cognito::UserPool', gen2UserPoolId, gen2Region);
+  gen2Resource = removeNestedJsonKeys(gen2Resource, ['ProviderURL', 'ProviderName', 'UserPoolId', 'Arn']);
+  // TODO: remove below line after EmailMessage, EmailSubject, SmsMessage, SmsVerificationMessage, EmailVerificationMessage, EmailVerificationSubject, AccountRecoverySetting inconsistency is fixed
+  gen2Resource = removeNestedJsonKeys(gen2Resource, [
+    'UserPoolTags',
+    'VerificationMessageTemplate.EmailMessage',
+    'VerificationMessageTemplate.SmsMessage',
+    'VerificationMessageTemplate.EmailSubject',
+    'SmsVerificationMessage',
+    'EmailVerificationSubject',
+    'AccountRecoverySetting',
+    'EmailVerificationMessage',
+  ]);
+  expect(gen2Resource).toEqual(gen1Resource);
+}
+
+function removeNestedJsonKeys<T extends Record<string, unknown>>(obj: T, keysToRemove: string[]): T {
+  const result = { ...obj };
+  keysToRemove.forEach((path) => {
+    const parts = path.split('.');
+    let current: Record<string, unknown> = result;
+    for (let i = 0; i < parts.length - 1; i++) {
+      const part = parts[i];
+      if (current[part] === undefined || current[part] === null) {
+        return; // Path doesn't exist, nothing to delete
+      }
+      current = current[part] as Record<string, unknown>;
+    }
+    const lastPart = parts[parts.length - 1];
+    if (current && lastPart in current) {
+      delete current[lastPart];
+    }
+  });
+  return result;
+}
```

Comment:
I think we have lodash in the node_modules.
if we do, consider using https://lodash.com/docs/4.17.15#unset instead.